modules/polling/api/fetchPollBy.ts

```diff
@@ -1,51 +1,54 @@
 import { config } from 'lib/config';
 import { fsCacheGet, fsCacheSet } from 'lib/fscache';
 import { markdownToHtml } from 'lib/markdown';
-import { gqlRequest } from 'modules/gql/gqlRequest';
-import { activePollByMultihash, activePollById } from 'modules/gql/queries/activePollBy';
+import { QueryFilterNames } from 'modules/gql/gql.constants';
+import { getQueryFilter } from 'modules/gql/gqlFilters';
+// import { gqlRequest } from 'modules/gql/gqlRequest';
+// import { activePollByMultihash, activePollById } from 'modules/gql/queries/activePollBy';
 import { SupportedNetworks } from 'modules/web3/constants/networks';
-import { networkNameToChainId } from 'modules/web3/helpers/chain';
+// import { networkNameToChainId } from 'modules/web3/helpers/chain';
 import { spockPollToPartialPoll } from '../helpers/parsePollMetadata';
 import { Poll } from '../types';
 import { PollSpock } from '../types/pollSpock';
 import { fetchPollMetadata } from './fetchPollMetadata';
-import { fetchAllSpockPolls } from './fetchPolls';
+import { fetchSpockPolls } from './fetchPolls';
 
 export async function fetchSpockPollById(
   pollId: number,
   network: SupportedNetworks
 ): Promise<PollSpock | undefined> {
-  const polls = await fetchAllSpockPolls(network);
-  return polls.find(poll => poll.pollId === pollId);
+  const filter = getQueryFilter(QueryFilterNames.PollId, pollId);
+  const [poll] = await fetchSpockPolls(network, filter);
+  return poll;
```

Comment:
what is the rationale to remove a query that can filter by multihash and now querying al the polls again?

Reply:
@rafinskipg I'm seeing two different things here. The query added above here `const [poll] = await fetchSpockPolls(network, filter);` fetches only one poll using the poll id. It uses the `PollId` query filter. The filter works exactly the same way as the multihash filter below where we do:
```
  variables: {
      argPollMultihash: `${slug}%`
    }
```
The only difference is I've moved the filters to their own file and defined it explicitly.

Below I'm not sure why the code is commented out, but I suggest we use the new filter pattern I added for gql queries. It would just be a matter of moving the `argPollMultihash: `${slug}%` variable into the gqlFilters file.